ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Query.hs

```diff
@@ -162,6 +165,18 @@ getIsInvalidBlock
 getIsInvalidBlock CDB{..} =
   fmap (fmap (fmap invalidBlockReason) . flip Map.lookup) <$> readTVar cdbInvalid
 
+getIsValid
+  :: forall m blk. (IOLike m, HasHeader blk)
+  => ChainDbEnv m blk
+  -> STM m (RealPoint blk -> Maybe Bool)
+getIsValid CDB{..} = do
+    prevApplied <- LgrDB.getPrevApplied cdbLgrDB
+    invalid     <- forgetFingerprint <$> readTVar cdbInvalid
+    return $ \pt@(RealPoint _ hash) ->
+      if | Set.member pt prevApplied -> Just True
+         | Map.member hash invalid   -> Just False
```

Comment:
Swap order, document why and add a TODO to add an invariant that it is in either one or the other. Create issue (mention why the tests are not catching this, i.e., because of #2232).